src/main/java/com/teragrep/pth10/steps/teragrep/TeragrepBloomStep.java

```diff
@@ -162,32 +158,35 @@ private Dataset<Row> estimateSize(Dataset<Row> dataset) {
     }
 
     public Dataset<Row> aggregate(Dataset<Row> dataset) {
-
-        FilterTypes filterTypes = new FilterTypes(this.zeppelinConfig);
-
-        BloomFilterAggregator agg = new BloomFilterAggregator(inputCol, estimateCol, filterTypes.sortedMap());
-
+        final FilterTypes filterTypes = new FilterTypes(this.zeppelinConfig);
```

Comment:
I would have this FilterTypes object as a class variable rather than the zeppelinConfig. Realistically though, if thinking with single responsibility principle in mind, this object should be four objects. Now it's resposnsible for creating, updating, estimating and aggregating (like in the Enum). If it was refactored into multiple objects, the FilterTypes as class variable suggestion would make more sense too. If that sounds good, I would open a new issue and eventually another PR for that so this one doesn't get too big.

Reply:
Good suggestion I am looking into it but I also created and issue about splitting this object https://github.com/teragrep/pth_10/issues/374

Reply:
Moved FilterTypes and LazyConnection to class variables, number of class variables are high for now but refactoring is easier with this change. 
Refactoring will also lower the number of variables of this class.